### ISSUE TITLE
Mixin crash; replace Redirect with Injection

### ISSUE THREAD

**Issue by Mg138** [CafeteriaGuild/DeepMobLearning-Refabricated#122]
```
org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:278) ~[quilt-loader-0.17.11.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:153) ~[quilt-loader-0.17.11.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:156) ~[quilt-loader-0.17.11.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_1792.method_7867(class_1792.java:74) ~[client-intermediary.jar:?]
	at net.minecraft.class_2248.method_8389(class_2248.java:554) ~[client-intermediary.jar:?]
	at net.minecraft.class_7143$class_7144.method_41583(class_7143.java:60) ~[client-intermediary.jar:?]
	at net.minecraft.class_7143$class_7144.method_41581(class_7143.java:66) ~[client-intermediary.jar:?]
	at net.minecraft.class_7143.method_41579(class_7143.java:199) ~[client-intermediary.jar:?]
	at net.minecraft.class_5458.method_44104(class_5458.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_5458.method_30566(class_5458.java:105) ~[client-intermediary.jar:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) [?:?]
	at net.minecraft.class_5458.<clinit>(class_5458.java:104) [client-intermediary.jar:?]
	at net.minecraft.class_2378.<clinit>(class_2378.java:326) [client-intermediary.jar:?]
	at net.minecraft.class_2966.method_12851(class_2966.java:50) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:161) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) [quilt-loader-0.17.11.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76) [quilt-loader-0.17.11.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:24) [quilt-loader-0.17.11.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) [NewLaunch.jar:?]
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector redirectPullProgress(ILnet/minecraft/class_1799;Lnet/minecraft/class_1937;Lnet/minecraft/class_1309;I)F in quilt_item_extension.mixins.json:bow.BowItemMixin from mod quilt_item_extension failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap item_extension.refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	... 29 more
```

Full log: https://mclo.gs/CdlOTS8

**Comment by Luligabi1** [comment]
Quilt is currently not supported.

**Comment by Mg138** [comment]
Could you not use an Inject instead? Using a Redirect there seems excessive and it being a Quilt conflict is not a good excuse not to fix it.

**Comment by Mg138** [comment]
Dunno how notifications work, tagging @Luligabi1 just to be sure.

**Comment by Mg138** [comment]
We could make a PR, if one would accept it.

**Comment by ZestyBlaze** [comment]
It's more a point of that Luli won't fix it because the mod isn't meant to support Quilt, Luli doesn't actively seek out to fix things for Quilt so he isn't going to go out of the way to fix it

**Comment by Luligabi1** [comment]
> Could you not use an Inject instead? Using a Redirect there seems excessive and it being a Quilt conflict is not a good excuse not to fix it.

@Mg138 I can see how this could cause issues with other mods, so I'll add it to my backlog. Do note that since it's not causing issues even on big kitchensink modpacks, it'll take a ***looooong*** while to get to it. If you're in a hurry, I'd merge a PR fixing it.

**Comment by Mg138** [comment]
Aight, glad to hear that!

**Comment by Luligabi1** [comment]
Fixed on 1.20 branch
